packages/svelte/src/compiler/phases/3-transform/client/visitors/template.js

```diff
@@ -959,6 +960,16 @@ function serialize_bind_this(bind_this, context, node) {
 	return b.call('$.bind_this', ...args);
 }
 
+/**
+ * @param {import("estree").Expression[]} args
+ * @param {import("../types.js").ComponentClientTransformState} state
+ */
+function push_dev_inspector_data(args, state) {
+	if (state.options.dev && state.options.filename) {
+		args.push(b.literal(state.options.filename), b.literal('sloc' + hash(state.options.filename)));
+	}
```

Comment:
Couldn't this also contain the 
```js
			/** @type {import('estree').Expression[]} */
			const args = [id];

			push_dev_inspector_data(args, state);

			close = b.stmt(b.call('$.append', b.id('$$anchor'), ...args));
```
part? From a quick look it seems like those are also exactly the same between usage sites. i.e. final method is

```js
function push_dev_inspector_data(args, state) {
	const args = [id];
	if (state.options.dev && state.options.filename) {
		args.push(b.literal(state.options.filename), b.literal('sloc' + hash(state.options.filename)));
	}
	return b.stmt(b.call('$.append', b.id('$$anchor'), ...args));
}
```

Reply:
I didn't want to include the append bit as that was unrelated to inspector logic. I don't think it needs to be that DRY?

Reply:
Oops I missed renaming the function to something that fits better in my suggestion, like `construct_append_method`